### PR TITLE
Change 'm' to 's' and improve message when externals are dirty

### PR DESCRIPTION
**Pull request by billsacks**
(1) Change 'm' to 's' to denote out-of-sync: I found it hard to remember
    what was meant by 'm' vs. 'M'. Hopefully it will be easier to
    remember 's' vs. 'M'.

(2) Improve the message when checkout_externals fails to run because one
    or more externals are in M state.

User interface changes?: Yes
Changed 'm' to 's'

Fixes: Partially addresses NCAR/manage_externals#51

Testing:
  test removed:
  unit tests: pass
  system tests: pass
  manual testing: Checked status and help output

### CODE REVIEW

**Comment by billsacks** [comment]
@bandre-ucar can you review this?

I made this PR off of the ignore-untracked-files branch because it was somewhat tied in with the changes there. But I can rebase it to be off of master if you'd prefer.

File: manic/externals_status.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class ExternalStatus(object):
     DEFAULT = '-'
     UNKNOWN = '?'
     EMPTY = 'e'
-    MODEL_MODIFIED = 'm'
+    MODEL_MODIFIED = 's'  # a.k.a. out-of-sync
```
Comment (billsacks) [inline manic/externals_status.py:33]: I noticed that 's' is also used for 'STANDALONE'. I wasn't sure how, if at all, STANDALONE is used, so I wasn't sure what, if anything, to do about this. @bandre-ucar do you have thoughts on this?

**Comment by bandre-ucar** [comment]
I think standalone is used to identify the standalone component in standalone checkouts to indicate that it isn't managed by checkout_externals.

File: manic/checkout.py [hunk 2]
```diff
@@ -249,7 +250,8 @@ def main(args):
     Parse externals file and load required repositories or all repositories if
     the --all option is passed.
     """
-    logging.info('Begining of checkout_externals')
+    progname = 'checkout_externals'
```
Comment (bandre-ucar) [inline manic/checkout.py]: This should probably be:
```
program_name = sys.argv[0]
```

File: manic/checkout.py [hunk 3]
```diff
@@ -281,12 +283,21 @@ def main(args):
...
+The following are two options for how to proceed:
+
+(1) Go into each external that is not in a clean state and issue either
+    an 'svn status' or a 'git status' command. Either revert or commit
+    your changes so that all externals are in a clean state. (Note,
+    though, that it is okay to have untracked files in your working
+    directory.) Then rerun {progname}.
+
+(2) Alternatively, you do not have to rely on {progname}. Instead, you
+    can manually update out-of-sync externals (labeled with 's' above)
```
Comment (bandre-ucar) [inline manic/checkout.py]: Is it worth replacing 'M' and 's' with {modified} and {out_of_sync}?

**Comment by billsacks** [comment]
> I think standalone is used to identify the standalone component in standalone checkouts to indicate that it isn't managed by checkout_externals.

So is it a problem to use 's' for both, or is this irrelevant because (it seems) the 's' for standalone is never displayed to the user?

File: manic/checkout.py (hunk 2 above)
Comment (billsacks) [reply manic/checkout.py]: Fixed in commit I just pushed

File: manic/checkout.py (hunk 3 above)
Comment (billsacks) [reply manic/checkout.py]: Sorry, I'm not clear on what you're suggesting here; can you clarify?

**Comment by billsacks** [comment]
@bandre-ucar just want to make sure this one doesn't fall through the cracks. (I had trouble finding it myself. Lesson learned that it's a pain to not have PRs in the main repo.)

**Comment by billsacks** [comment]
Once @bandre-ucar approves this, I may need to put the changes on the appropriate branch: I'm not sure that simply merging the PR will be enough at this point, given the complicated branch history.

**Comment by bandre-ucar** [comment]
@billsacks Sorry, I didn't realize this PR was still open on your fork. I've only been checking the main repo and my fork.

I'm fine with these changes. I'm aware that this branch exists and needs to come in with the untracked files branch.

**Comment by billsacks** [comment]
Thanks, @bandre-ucar . I understand from your comment that you'll do the necessary merge; let me know if I'm misinterpreting and you'd like me to do something more here.

**Comment by bandre-ucar** [comment]
@billsacks Correct, I'll will take care of any additional work getting this to master.

**Comment by bandre-ucar** [comment]
This will go to master in: https://github.com/NCAR/manage_externals/pull/65

**Pull request closed without merging**